airbrakes/data_handling/apogee_predictor.py

```diff
@@ -188,12 +213,38 @@ def _prediction_loop(self) -> None:
                 self._current_altitude = data_packet.current_altitude
                 self._current_velocity = data_packet.vertical_velocity
 
-            # TODO: play around with this value
-            # Run apogee prediction every 100 data points:
             if len(self._accelerations) - last_run_length >= APOGEE_PREDICTION_FREQUENCY:
+                # We only want to keep curve fitting if the curve fit hasn't converged yet
+                if not self._has_apogee_converged():
+                    curve_coefficients = self._curve_fit()
+                    lookup_table = self._generate_prediction(curve_coefficients)
                 self._cumulative_time_differences = np.cumsum(self._time_differences)
-                params = self._curve_fit()
-                self._apogee_prediction_value.value = self._get_apogee(params)
+                # Predicts the apogee using the lookup table and linear interpolation
+                predicted_apogee = np.interp(
+                    self._current_velocity, lookup_table[0], lookup_table[1]
+                )
+                self._apogee_prediction_value.value = predicted_apogee
+                self._predicted_apogees = np.append(self._predicted_apogees, predicted_apogee)
                 last_run_length = len(self._accelerations)
             # notifies tests that prediction is complete
             self._prediction_complete.set()
+
+    def _has_apogee_converged(self) -> bool:
+        """
+        Checks if the last 5 apogees are within 3% of each other.
+        :return: True if the last 5 apogees are within 3% of each other, otherwise False.
+        """
+        if len(self._predicted_apogees) < NUMBER_OF_PREDICTIONS:
+            return False  # Not enough data to check convergence
+
+        # Because we return 0.0 if apogee hasn't been predicted, we don't want to say it's
+        # converged if there's any 0s
+        if np.any(self._predicted_apogees == 0):
+            return False
+
+        avg_apogee = np.mean(self._predicted_apogees[-NUMBER_OF_PREDICTIONS:])
+
+        # Check if each apogee is within 3% of the average
+        return bool(
```

Comment:
it's already a bool, no need of bool()

Reply:
lol thats what I thought but pycharm was mad that it was a np.bool or some shit